streams/catalog/src/main/java/com/hortonworks/streamline/streams/catalog/processor/CustomProcessorInfo.java

```diff
@@ -161,6 +174,16 @@ public void setTopologyComponentUISpecification(ComponentUISpecification compone
         this.topologyComponentUISpecification = componentUISpecification;
     }
 
+    @JsonIgnore
+    public String getDigest() {
+        return digest;
+    }
+
+    @JsonIgnore
+    public void setDigest (String digest) {
+        this.digest = digest;
+    }
+
     public CustomProcessorInfo fromTopologyComponentBundle (TopologyComponentBundle topologyComponentBundle) throws IOException {
```

Comment:
You may want to refactor this to factory method instead of instance method and have a private no args constructor to control on how this instance is created. One public/package constructor with the required fields can be declared to create an instance from tests or other scenarios.

Reply:
@satishd This object is also used in json serialization and deserialization during HTTP requests. Not sure if a private constructor will work with that or not. Also, a public constructor with required fields will make it a big constructor with lot of arguments. Unless you see a significant advantage of using a static factory method over instance method, I would like to keep  it as it is since that change involves testing everything all over again. We can revisit later if needed.

Reply:
@priyank5485 private constructor works fine with json serialization as it is done in other places in streamline and registry. 
Having static factory method will not allow you to create an empty instance and get into issues like [here](https://github.com/hortonworks/streamline/pull/983/files/728c371e0b51c9028bf2c0f7d10646f2fcef3afc#diff-6d94aae217fe8e21348f49c18e1ba329L1091). I m fine if we want to revisit this later.

Reply:
Ok. I will made a note of it and take it up as part of other CP enhancements for next release.